commands/http/handler.go

```diff
@@ -121,6 +121,16 @@ func (i internalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	i.ctx.Context = ctx
 	req.SetContext(i.ctx)
 
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func() {
+			select {
+			case <-cn.CloseNotify():
+				cancel()
+			case <-ctx.Done():
+			}
+		}()
+	}
+
```

Comment:
That is odd...
1. (client) http request conn _goes away_
2. (server) CloseNotifier notifies, canceling context.
3. (server) writes out an error, or cuts writing short (cancelled context)
4. (client) (which had theoretically gone away!?) receives clipped output
5. (client) errors out.